index.js

```diff
@@ -1,17 +1,56 @@
 module.exports = leftpad;
 
-function leftpad (str, len, ch) {
-  str = String(str);
-
-  var i = -1;
-
-  if (!ch && ch !== 0) ch = ' ';
-
-  len = len - str.length;
-
-  while (++i < len) {
-    str = ch + str;
+if (!String.prototype.repeat) {
+  String.prototype.repeat = function(count) {
+    'use strict';
+    if (this == null) {
+      throw new TypeError('can\'t convert ' + this + ' to object');
+    }
+    var str = '' + this;
+    count = +count;
+    if (count != count) {
+      count = 0;
+    }
+    if (count < 0) {
+      throw new RangeError('repeat count must be non-negative');
+    }
+    if (count == Infinity) {
+      throw new RangeError('repeat count must be less than infinity');
+    }
+    count = Math.floor(count);
+    if (str.length == 0 || count == 0) {
+      return '';
+    }
+    // Ensuring count is a 31-bit integer allows us to heavily optimize the
+    // main part. But anyway, most current (August 2014) browsers can't handle
+    // strings 1 << 28 chars or longer, so:
+    if (str.length * count >= 1 << 28) {
+      throw new RangeError('repeat count must not overflow maximum string size');
+    }
+    var rpt = '';
+    for (;;) {
```

Comment:
I am not very frontendy but, which is the purpose of a for(;;), it is not very readable.
First thing you expect when you read a code and you find a for is to have the information about the loop, which variable is going to hold the count (because it's a for), what is the stop condition and how are you going to increment the count.
On a for(;;) you don't have any information about the loop, in fact you know you are going to face an infinite loop and just pray something inside is going to make it stop.
There are different ways of looping and each one has a purpose, probably you can find exception or situation where a classic loop does not fit, but it's hard to me think a situation where a for(;;) it is acceptable, if you want and infinite loop, you can use while(true), but again, throw an infinite loop is rarely an acceptable solution.